corehq/apps/cloudcare/static/cloudcare/js/form_entry/utils.js

```diff
@@ -149,5 +149,28 @@ hqDefine("cloudcare/js/form_entry/utils", function () {
         return broadcastObj;
     };
 
+    /**
+     * Gets the root container of a question.
+     * @param {Object} question - The Question Object
+     * @param {Function} predicate - Optional. If provided, stop when this condition is reached.
+     * If not provided, the root Form object will be returned.
+    **/
+    module.getRootContainer = (question, stopCallback) => {
```

Comment:
Interested in feedback on readability here. I debated just doing one `getRootForm` function and a separate function to get the nearest repeat group / form ancestor.

Reply:
Ended up splitting this.